app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,33 @@
+class FoodsController < ApplicationController
+  load_and_authorize_resource
+
+  def index
+    @foods = current_user.foods
+  end
+
+  def new; end
+
+  def create
+    new_food = current_user.foods.new(food_params)
+    if new_food.save
+      flash[:success] = 'You successfully added a food item'
+      redirect_to foods_url
+    else
+      flash.now[:error] = 'Error: You could not add a food item'
+      render :new
+    end
```

Comment:
- [ ] Unable to add food details successfully. When I enter food details, such as the sample details as seen in the screenshot, and then click on save, a flash message is displayed saying, _You could not add a food item_ and entries are not added to food table. Kindly ensure to improve upon this aspect.

Kindly note, the **next code reviewer(s)** will be able to review other aspects of the requirement if you get this preliminary step working, and therefore there might be new issues highlighted in upcoming reviews.

**Error:**
![image](https://user-images.githubusercontent.com/12745474/152407974-e41c67a8-9aaf-44ed-8466-26af201d17e1.png)

Reply:
Hi @rloterh, 

Thank you for the feedback